webpack/assets/javascripts/react_app/components/notifications/notifications.test.js

```diff
@@ -1,70 +1,70 @@
-import API from '../../redux/API/API';
-import { doesDocumentHasFocus } from '../../common/document';
-
+import thunk from 'redux-thunk';
+import IntegrationTestHelper from '../../common/IntegrationTestHelper';
+import notificationReducer from '../../redux/reducers/notifications';
 import { componentMountData, serverResponse } from './notifications.fixtures';
-
 import Notifications from './';
+import API from '../../redux/API/API';
+import { APIMiddleware } from '../../redux/API';
+import { NOTIFICATIONS } from '../../redux/consts';
+import { IntervalMiddleware } from '../../redux/middlewares';
+import { registeredIntervalException } from '../../redux/middlewares/IntervalMiddleware/IntervalHelpers';
+import { DEFAULT_INTERVAL } from '../../redux/actions/notifications/constants';
 
+jest.useFakeTimers();
 jest.mock('../../redux/API/API');
-jest.mock('../../common/document');
+jest.mock('../../redux/actions/notifications/constants', () => ({
+  DEFAULT_INTERVAL: 1,
+}));
 
-let failResponse = { response: { status: 200 } };
+const notificationProps = {
+  data: componentMountData,
+};
 
-function mockjqXHR() {
-  return {
-    then: callback => {
-      callback(JSON.parse(serverResponse));
-      return mockjqXHR();
-    },
-    catch: failCallback => {
-      failCallback(failResponse);
-      return mockjqXHR();
-    },
-  };
-}
+const configureIntegrationHelper = () => {
+  const reducers = { notifications: notificationReducer };
+  const middlewares = [thunk, IntervalMiddleware, APIMiddleware];
+  return new IntegrationTestHelper(reducers, middlewares);
+};
 
 describe('notifications', () => {
-  doesDocumentHasFocus.mockImplementation(() => true);
-
   beforeEach(() => {
     global.tfm = {
       tools: {
         activateTooltips: () => {},
       },
     };
-    API.get = mockjqXHR;
   });
 
   it('full flow', () => {
-    const wrapper = mount(
-      <Notifications data={componentMountData} store={generateStore()} />
-    );
-    wrapper.find('.fa-bell').simulate('click');
-    expect(wrapper.find('.panel-group')).toHaveLength(1);
-    wrapper.find('.panel-group .panel-heading').simulate('click');
-    expect(wrapper.find('.unread')).toHaveLength(1);
-    wrapper.find('.unread').simulate('click');
-    expect(wrapper.find('.unread')).toHaveLength(0);
-  });
-
-  it('should redirect to login when 401', () => {
-    window.location.replace = jest.fn();
-    failResponse = { response: { status: 401 } };
+    API.get.mockImplementation(async () => serverResponse);
 
-    mount(<Notifications data={componentMountData} store={generateStore()} />);
-    expect(global.location.replace).toBeCalled();
+    const testHelper = configureIntegrationHelper();
+    const wrapper = testHelper.mount(<Notifications {...notificationProps} />);
+    expect(setInterval).toHaveBeenCalledTimes(1);
+    expect(setInterval).toHaveBeenLastCalledWith(
+      expect.any(Function),
+      DEFAULT_INTERVAL
+    );
+    setTimeout(() => {
```

Comment:
can you just add a comment above `setTimeout` that clarifies this is a workaround?